src/encoded/static/components/award.js

```diff
@@ -126,11 +158,163 @@ function createDoughnutChart(chartId, values, labels, colors, baseSearchUri, nav
     });
 }
 
+function createBarChart(chartId, unreplicatedLabel, unreplicatedDataset, isogenicDataset, anisogenicDataset, colors, labels, replicatelabels, baseSearchUri, navigate) {
+    return new Promise((resolve) => {
+        require.ensure(['chart.js'], (require) => {
+            const Chart = require('chart.js');
+            const datasets = [{}, {}, {}];
+            if (unreplicatedDataset.some(x => x > 0)) {
+                datasets[0].label = 'unreplicated';
+                datasets[0].data = unreplicatedDataset;
+                datasets[0].backgroundColor = colors[0];
+                if (isogenicDataset.some(x => x > 0)) {
+                    datasets[1].label = 'isogenic';
+                    datasets[1].data = isogenicDataset;
+                    datasets[1].backgroundColor = colors[1];
+                    if (anisogenicDataset.some(x => x > 0)) {
+                        datasets[2].label = 'anisogenic';
+                        datasets[2].data = anisogenicDataset;
+                        datasets[2].backgroundColor = colors[2];
+                    }
+                } else if (isogenicDataset.every(x => x === 0) && anisogenicDataset.some(x => x > 0)) {
+                    datasets[1].label = 'anisogenic';
+                    datasets[1].data = anisogenicDataset;
+                    datasets[1].backgroundColor = colors[1];
+                }
+            } else if (unreplicatedDataset.every(x => x === 0) && isogenicDataset.some(x => x > 0)) {
+                datasets[0].label = 'isogenic';
+                datasets[0].data = isogenicDataset;
+                datasets[0].backgroundColor = colors[0];
+                if (anisogenicDataset.some(x => x > 0)) {
+                    datasets[1].label = 'anisogenic';
+                    datasets[1].data = anisogenicDataset;
+                    datasets[1].backgroundColor = colors[1];
+                }
+            } else if (unreplicatedDataset.every(x => x === 0) && isogenicDataset.every(x => x === 0) && anisogenicDataset.some(x => x > 0)) {
+                datasets[0].label = 'anisogenic';
+                datasets[0].data = anisogenicDataset;
+                datasets[0].backgroundColors = colors[0];
+            }
```

Comment:
While technically fine, I think there’s too many “magic” numbers here with the 0, 1, and 2 array indices. I’d prefer to use three different object variables, and then combine them into an array to pass in the `data` object when you create the chart below.